include/deal.II/base/vectorization.h

```diff
@@ -78,6 +79,37 @@ struct EnableIfScalar<VectorizedArray<Number> >
 };
 
 
+
+#ifndef DEAL_II_WITH_CXX11
+// Specify the types for the implemented multiplications explicitly
+
+template <typename Number>
+struct ProductType<Number, VectorizedArray<Number> >
+{
+  typedef VectorizedArray<Number> type;
+};
+
+template <typename Number>
+struct ProductType<VectorizedArray<Number>, Number>
+{
+  typedef VectorizedArray<Number> type;
+};
+
+template<>
+struct ProductType<double, VectorizedArray<float> >
+{
+  typedef VectorizedArray<float> type;
```

Comment:
No, that is not how `operator*` is implemented. From the [documentation](https://www.dealii.org/developer/doxygen/deal.II/classVectorizedArray.html#ab0a712f36a0a7e25d81b654e58f2e2ed)
> VectorizedArray< float > 	operator* (const VectorizedArray< float > &v, const double &u)
> VectorizedArray< float > 	operator* (const double &u, const VectorizedArray< float > &v)